### PR TITLE
Linus: basic setup for development

### PR DESCRIPTION
**Pull request by JohaJung**
This PR adds [documentation and setup instructions](https://github.com/TUDa-VLAOrga/VLAOrga/blob/LinusConnectionInterface/Linus/ReadmeLinus.md) to get a local istance of linus running.

It does not yet implement any connection inside our application's backend, is just for development and anlysis of how linus works.

Try out the setup instructions and play around! 

relevant issue: #40 

---

### Implementation

- [ ] Does this code change accomplish what it is supposed to do?
- [ ] Can this solution be simplified without unreasonable effort
- [ ] Is the code modular enough?
- [ ] Does similar functionality already exist in the codebase? If yes, why isn’t it reused?
- [ ] Single Responsibility Principle: Does any class serve only one purpose? If not, it should probably split up in multiple classes.
- [ ] Interface Segregation: If interface implemented, is it fully implemented?
- [ ] Dependency Injection: Are central object instantiation processes managed by a single context?
- [ ] If any functionality is missing, is it marked with a comment that starts with TODO:?
- [ ] Do you see any potential to improve the performance of the code significantly? 

### Logic Errors and Bugs

- [ ] Can you think of any use case in which the code does not behave as intended?
- [ ] Can you think of any inputs or external events that could break the code?
- [ ] Are edge cases covered by tests?

### Error Handling and Logging

- [ ] Is error handling done the correct way?
- [ ] Should any logging or debugging information be added or removed?
- [ ] Are error messages user-friendly?

### Dependencies (internal and external)

- [ ] Were updates to documentation, configuration, or readme files made as required by this change?
- [ ] Is the API.md up-to-date?
- [ ] Are the enumerations between frontend and backend synchronized?

### Security and Data Privacy

- [ ] Does the code introduce any security vulnerabilities?
- [ ] Is (user) input validated, sanitized, and escaped to prevent security attacks such as cross-site scripting or SQL injection?
- [ ] Is sensitive data like user data handled and stored securely?
- [ ] Is the right encryption used?
- [ ] Is data retrieved from external APIs or libraries checked for security issues?
- [ ] Were any secrets committed?

### Testing and Testability (later)

- [ ] Have automated tests been added, or have related ones been updated to cover the change?
- [ ] Do the existing tests reasonably cover the code change?
- [ ] Are there some test cases, input or edge cases that should be tested in addition?

### Usability and Accessibility

- [ ] Is the proposed solution (UI) accessible (e.g. colorblindness, ...)?
- [ ] Is the API/UI intuitive to use?
- [ ] Was the app tested on another device?

### Readability

- [ ] Is the code easy to understand?
- [ ] Which parts were confusing to you and why?
- [ ] Can the readability of the code be improved by smaller methods/different names?
- [ ] Is the code located in the right file/folder?
- [ ] Do you think certain methods should be restructured to have a more intuitive control flow?
- [ ] Are there redundant or outdated comments?
- [ ] Could some comments convey the message better?
- [ ] Could some comments be removed by making the code itself more readable?

### CODE REVIEW

File: Linus/ReadmeLinus.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Setup of Linus as development instance:
+
+1. Clone linus as submodule by running (inside the Linus/ directory)
+    
+        VLAOrga/Linus $ git submodule init datenbank-physik
+   If the datenbank-physik folder is still empty:
```
Comment (JohaJung) [inline Linus/ReadmeLinus.md]: @CarolinRoell this misses a "2." at the front, and probably also a newline before this line.

Comment (CarolinRoell) [reply Linus/ReadmeLinus.md]: I thought of it as a subpoint of 1, but this works as well

Comment (JohaJung) [reply Linus/ReadmeLinus.md]: Ah, then you (or probably GitHub editor) broke the numbering in code. It continues with 3.

The rendered view hides this and numbers correctly

**Comment by JohaJung** [comment]
This now includes instructions how to set up linus with a dummy user. (Inserting one line into the sql database), hopefully now everything is ready that we can implement a connection from our Java backend to this linus database.

File: VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/linus_connection/Appointment.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+
+/**
+ * Represents a lecture in a certain semester. For example "Physik I, WS 25/26".
+ */
+@Entity
+@Table(name = "reservation")
+public class Appointment {
```
Comment (JohaJung) [inline VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/linus_connection/Appointment.java]: Please use the same name for table and class.

Don't name it appointment, we'll use that for our own calendar events.

-> rename class to Reservation (or better Booking?)

File: VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/linus_connection/AppointmentRepository.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package de.vlaorgatu.vlabackend.linus_connection;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+/**
+ * Repository for accessing and managing {@link Appointment} entities.
+ */
+public interface AppointmentRepository extends JpaRepository<Appointment,Long> {
+}
```
[the remark this answers is not in the document]
Comment (CarolinRoell) [reply VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/linus_connection/AppointmentRepository.java]: Sollte durch die LinusAppointmentAutoConfiguration Klasse konfiguriert sein

File: VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/linus_connection/LinusExperimentBooking.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+/**
+ * Represents that an experiment is booked for an appointment.
+ */
+@Getter
+@Entity
+@Table(name = "reservation_experiment")
+public class LinusExperimentBooking {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
```
[the remark this answers is not in the document]
Comment (CarolinRoell) [reply VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/linus_connection/LinusExperimentBooking.java]: The id annotation implies this: " If no Column annotation is specified, the primary key column name is assumed to be the name of the primary key property or field."
If we like it more, we can specify it for readability.

File: VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/linusconnection/LinusAppointmentController.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+
+/**
+ * REST endpoints for managing {@link LinusAppointment} entities.
+ */
+@Controller
+@RequestMapping("/api/linus/appointment")
```
Comment (JohaJung) [inline VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/linusconnection/LinusAppointmentController.java]: When merging with #69 , adjust to use base path from application.properties (see comment at ExperimentBookingController below).

**Comment by JohaJung** [comment]
Great, it works! :heart:

File: VLA-Backend/src/main/resources/application.properties [hunk 6]
```diff
@@ -1,11 +1,16 @@
 spring.application.name=VLA-Backend
-spring.datasource.url=${SPRING_DATASOURCE_URL}
-spring.datasource.username=${SPRING_DATASOURCE_USERNAME}
-spring.datasource.password=${SPRING_DATASOURCE_PASSWORD}
-spring.datasource.driver-class-name=org.postgresql.Driver
+spring.datasource.jdbcUrl = ${SPRING_DATASOURCE_URL}
+spring.datasource.username = ${SPRING_DATASOURCE_USERNAME}
+spring.datasource.password = ${SPRING_DATASOURCE_PASSWORD}
```
Comment (MarcoLeipnitz) [inline VLA-Backend/src/main/resources/application.properties]: Affected by potential name changes, see .services

File: VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/linusconnection/LinusAppointment.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    private String orderTime;
+
+    /**
+     * Status of the appointment.
+     */
+    @Column(name = "status")
+    private Long status;
+
+    /**
+     * Date and time when the appointment takes place.
```
Comment (MarcoLeipnitz) [inline VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/linusconnection/LinusAppointment.java]: Consider adding a definition of how the string is stored
ISO String? UTC String? ...

File: VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/linusconnection/LinusAppointment.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+     * Date and time when the reservation of the appointment was made in linus.
+     */
+    @Column(name = "ordertime")
+    private String orderTime;
+
+    /**
+     * Status of the appointment.
+     */
+    @Column(name = "status")
+    private Long status;
```
Comment (MarcoLeipnitz) [inline VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/linusconnection/LinusAppointment.java]: I suppose the status is represented as a general long while being mapped to some other value.
Consider an enum or final static attribute class to make this easier to work with.

File: docker-compose.services.yml [hunk 9]
```diff
@@ -31,8 +38,12 @@ services:
       SPRING_DATASOURCE_URL: ${SPRING_DATASOURCE_URL}
       SPRING_DATASOURCE_USERNAME: ${SPRING_DATASOURCE_USERNAME}
       SPRING_DATASOURCE_PASSWORD: ${SPRING_DATASOURCE_PASSWORD}
+      SPRING_SECOND_DATASOURCE_URL: ${SPRING_SECOND_DATASOURCE_URL}
+      SPRING_SECOND_DATASOURCE_USERNAME: ${SPRING_SECOND_DATASOURCE_USERNAME}
+      SPRING_SECOND_DATASOURCE_PASSWORD: ${SPRING_SECOND_DATASOURCE_PASSWORD}
```
[the remark this answers is not in the document]
Comment (JohaJung) [reply docker-compose.services.yml]: \* VLAOrga and Linus

File: VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/linusconnection/LinusAppointment.java [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+     */
+    // TODO: possible candidate for @JoinColumn
+    @Column(name = "user_id")
+    private Long linusUserId;
+
+    /**
+     * Date and time when the reservation of the appointment was made in linus.
+     */
+    @Column(name = "ordertime")
+    private String orderTime;
```
Comment (JohaJung) [inline VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/linusconnection/LinusAppointment.java]: I just saw that this originally is a Datetime. We should read it as such.

See https://git.rwth-aachen.de/datenbank-physik/datenbank-physik/-/blob/master/src/Entity/Reservation.php#L36-L39

File: VLA-Backend/src/test/java/de/vlaorgatu/vlabackend/TestcontainersConfiguration.java [hunk 11]
```diff
@@ -15,4 +17,23 @@ PostgreSQLContainer<?> postgresContainer() {
...
+        return (properties) -> {
+            // Connect our Spring application to our Testcontainers instances
+            properties.add("spring.second-datasource.jdbcUrl", mariadbContainer::getJdbcUrl);
+            properties.add("spring.second-datasource.username", mariadbContainer::getUsername);
+            properties.add("spring.second-datasource.password", mariadbContainer::getPassword);
+            properties.add("spring.datasource.jdbcUrl", postgreSqlContainer::getJdbcUrl);
+            properties.add("spring.datasource.username", postgreSqlContainer::getUsername);
+            properties.add("spring.datasource.password", postgreSqlContainer::getPassword);
+        };
+    }
```
[the remark this answers is not in the document]
Comment (CarolinRoell) [reply VLA-Backend/src/test/java/de/vlaorgatu/vlabackend/TestcontainersConfiguration.java]: Added as TODO
